### PR TITLE
Simplify rand seeding

### PR DESCRIPTION
**Pull request by ash2k**
**What this PR does / why we need it**:
Simplify/cleanup random number generator seeding and usage. Please see descriptions of individual commits.

**Release note**:
```release-note
NONE
```
/kind cleanup
/sig api-machinery

### CODE REVIEW

**Review by dims — APPROVED** [review]
/lgtm
/approve

File: pkg/volume/flocker/flocker_util.go [hunk 1]
```diff
@@ -68,7 +67,6 @@ func (util *FlockerUtil) CreateVolume(c *flockerVolumeProvisioner) (datasetUUID
 	}
 
 	// select random node
-	rand.Seed(time.Now().UTC().UnixNano())
```
Comment (mikedanese) [inline pkg/volume/flocker/flocker_util.go]: ...wat...

**Review by mikedanese — COMMENTED** [review]
It seems like we could do a little better initializing rand consistently in e2e tests but this all looks good.

/lgtm
/approve

File: pkg/volume/flocker/flocker_util.go (hunk 1 above)
Comment (logicalhan) [reply pkg/volume/flocker/flocker_util.go]: i believe that file uses the api-machinery rand, so this line should be able to be safely removed.

Comment (mikedanese) [reply pkg/volume/flocker/flocker_util.go]: I agree it is safe to remove. I'm more confused how it existed in the first place. We are reseeding a global rand every call to this function. I'm surprised this wasn't more problematic.

Comment (ash2k) [reply pkg/volume/flocker/flocker_util.go]: I think I saw two more weird places like this, but they were in vendor. One in aws-go-sdk and another one in some azure thing IIRC.

**Pull request merged**